statchatui/src/App/App.jsx

```diff
@@ -6,13 +6,78 @@ import SignUp from "../SignUp/SignUp";
 import SignIn from "../SignIn/SignIn";
 import Home from "../Home/Home";
 import Navbar from "../Navbar/Navbar";
+import Compare from "..//Compare//Compare";
 import { appData } from "../Data/index";
-
 import "./App.css";
 
+const GAME_NAMES = {
+  APEX: "Apex Legends",
+  R6: "Rainbow Six Siege",
+};
+
 function App() {
   const [isSignedIn, setIsSignedIn] = useState(false);
   const [data, setData] = useState(appData);
+  const [signedInUserData, setSignedInUserData] = useState({
+    gamertag: "",
+    platform: "",
+    gameName: "",
+  });
+  useEffect(() => {
+    //i set the signed in userdata to the local storage because when i refresh the compare component, it will be deleted unless i save it here
+    localStorage.setItem("signedInUserData", JSON.stringify(signedInUserData));
+  }, [signedInUserData]);
+  async function fetchStats(formData) {
+    const apexApiKey = process.env.REACT_APP_API_KEY;
+    const baseUrl = "https://api.mozambiquehe.re/bridge?auth=";
+    if (
+      formData.gamertag === "" ||
+      formData.platform === "" ||
+      formData.gameName === ""
+    ) {
+      return null;
+    }
+    try {
+      let response;
+      if (formData.gameName === GAME_NAMES.APEX) {
+        response = await fetch(
+          `${baseUrl}${apexApiKey}&player=${formData.gamertag}&platform=${formData.platform}`
+        );
+      } else if (formData.gameName === GAME_NAMES.R6) {
+        response = await fetch(
+          `https://api.henrikdev.xyz/r6/v1/profile/${formData.platform.toLowerCase()}/${
+            formData.gamertag
+          }`
+        );
+      }
+      const data = await response.json();
+      return data;
+    } catch (error) {
+      console.error("Error submitting form:", error);
+    }
+  }
+  useEffect(() => {
+    async function fetchData() {
```

Comment:
just to be more specific, maybe we can rename this to `fetchInitialData`?

It'll also be great if you could move this to the *module level* like you did with the original fetchStats function. This means pulling out code like `onStatsChange`. Notice that doing this with `fetchStats` the other day made it super easy for us to move it.